### PR TITLE
`Development`: Fix flaky playwright e2e test

### PR DESCRIPTION
**Pull request by sarpsahinalp**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

The flaky playwright tests Quiz Exercise Management › Quiz Exercise Creation › Creates a Quiz with Drag and Drop and Quiz Exercise Participation › DnD Quiz participation › Student can participate in DnD Quiz was mentioned in the last meeting to be fixed.

### Description

Added drag functionality which was not working before in the utils.

### Steps for Testing

Prerequisites:
- Local playwright setup

1. Run the mentioned playwright tests

If no local setup take a look at the playwright tests that passed in the build.

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by az108 — APPROVED** [review]
Good fix, discussed together already

File: src/test/playwright/support/pageobjects/exercises/quiz/QuizExerciseCreationPage.ts [hunk 1]
```diff
@@ -71,11 +77,9 @@ export class QuizExerciseCreationPage {
 
     async uploadDragAndDropBackground() {
         const fileChooserPromise = this.page.waitForEvent('filechooser');
-        const fileUploadPromise = this.page.waitForResponse(`${BASE_API}/fileUpload*`);
-        await this.page.locator('#background-image-input-form').click();
+        await this.page.locator('#background-file-input-button').click();
         const fileChooser = await fileChooserPromise;
-        await fileChooser.setFiles('exercise/quiz/drag_and_drop/background.jpg');
-        await fileUploadPromise;
+        await fileChooser.setFiles('../cypress/fixtures/exercise/quiz/drag_and_drop/background.jpg');
```
Comment (muradium) [inline src/test/playwright/support/pageobjects/exercises/quiz/QuizExerciseCreationPage.ts:82]:
```suggestion
        await fileChooser.setFiles('./fixtures/exercise/quiz/drag_and_drop/background.jpg');
```

File: src/test/playwright/support/pageobjects/exercises/quiz/QuizExerciseCreationPage.ts [hunk 2]
```diff
@@ -46,17 +46,23 @@ export class QuizExerciseCreationPage {
         await this.page.locator('#drag-and-drop-question-title').fill(title);
 
         await this.uploadDragAndDropBackground();
-        await this.page.mouse.move(50, 50);
+        const element = await this.page.$('.background-area');
```
Comment (muradium) [inline src/test/playwright/support/pageobjects/exercises/quiz/QuizExerciseCreationPage.ts:49]:
```suggestion
        const element = await this.page.locator('.background-area');
```

Locator-based methods are preferred

File: src/test/playwright/support/pageobjects/exercises/quiz/QuizExerciseCreationPage.ts [hunk 3]
```diff
@@ -46,17 +46,23 @@ export class QuizExerciseCreationPage {
...
+        const boundingBox = await element?.boundingBox();
+
+        if (!boundingBox) {
+            throw new Error('Could not get bounding box of element');
+        }
```
Comment (muradium) [inline src/test/playwright/support/pageobjects/exercises/quiz/QuizExerciseCreationPage.ts:54]:
```suggestion
        expect(boundingBox, { message: 'Could not get bounding box of element' }).not.toBeNull();
```
Assertions with `expect()` give more details when they fail. But please note that this way `boundingBox` is not unwrapped anymore, so a few small changes might be needed in following lines.

**Review by muradium — CHANGES_REQUESTED** [review]

**Review by muradium — COMMENTED** [review]
Tests pass locally. Thanks for fixing 👍

**Review by muradium — APPROVED** [review]
Thanks for changes. Code LGTM now

**Review by az108 — APPROVED** [review]
reapprove

**Pull request merged**